pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -270,11 +270,15 @@ func (s *FlinkStateMachine) handleClusterStarting(ctx context.Context, applicati
 	}
 
 	// Wait for all to be running
-	ready, err := s.flinkController.IsClusterReady(ctx, application)
+	clusterReady, err := s.flinkController.IsClusterReady(ctx, application)
 	if err != nil {
 		return statusUnchanged, err
 	}
-	if !ready {
+
+	// ignore the error, we just care whether it's ready or not
+	serviceReady, _ := s.flinkController.IsServiceReady(ctx, application, flink.HashForApplication(application))
```

Comment:
Since you do this here, you do not need to check `IsServiceReady` in SubmittingJob state right?

Reply:
Yeah, that check is probably not necessary any more

Reply:
Removed the check in SubmittingJob